compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorToGPU.cpp

```diff
@@ -266,7 +318,27 @@ struct LLVMGPUVectorToGPUPass
     createAsyncGroups(funcOp);
 
     if (llvmgpuUseMMASync) {
+
+      SmallVector<Operation*> opsToErase;
+      do {
+      opsToErase.clear();
+      funcOp.walk([&](Operation* op) {
+        if (isOpTriviallyDead(op))
+          opsToErase.push_back(op);
+      });
+      for(Operation* op : opsToErase)
+        op->erase();
+      } while(!opsToErase.empty());
+
```

Comment:
this is just a hack to remove dead operations before doing the unrolling. Otherwise the swizzling doesn't work  as the old vector operations are still there in the code (but dead). This is quite hacky. We might want to have the swizzling in a separate pass and run DCE in between